File: ruby/lib/ccpayment/services/checkout_service.rb
module CCPayment
  module Services
    class CheckoutService
      def initialize(client)
        @client = client
      end

      def create_checkout_url(data)
        @client.post('/createCheckoutUrl', data)
      end

      def hosted_order_info(order_id)
        @client.post('/hostedOrderInfo', { orderId: order_id })
      end

      def hosted_order_info_first(order_id)
        @client.post('/hostedOrderInfoFirst', { orderId: order_id })
      end

      def create_hosted_order_deposit(data)
        @client.post('/createHostedOrderDeposit', data)
      end

      def get_hosted_coin_usdt_price(coin_ids)
        @client.post('/getHostedCoinUSDTPrice', { coinIds: coin_ids })
      end

      def get_main_coin_list
        @client.post('/getMainCoinList', {})
      end

      def create_app_demo_order_deposit(data)
        @client.post('/createAppDemoOrderDeposit', data)
      end

      def confirm_trade(data)
        @client.post('/confirmTrade', data)
      end
    end
  end
end

File: java/src/main/java/com/ccpayment/services/CheckoutService.java
package com.ccpayment.services;

import com.ccpayment.APIError;
import com.ccpayment.Client;
import com.google.gson.JsonObject;

import java.io.IOException;

public class CheckoutService {
    private final Client client;

    public CheckoutService(Client client) {
        this.client = client;
    }

    public JsonObject createCheckoutUrl(JsonObject data) throws APIError, IOException {
        return client.post("/createCheckoutUrl", data, JsonObject.class);
    }

    public JsonObject hostedOrderInfo(String orderId) throws APIError, IOException {
        JsonObject data = new JsonObject();
        data.addProperty("orderId", orderId);
        return client.post("/hostedOrderInfo", data, JsonObject.class);
    }

    public JsonObject hostedOrderInfoFirst(String orderId) throws APIError, IOException {
        JsonObject data = new JsonObject();
        data.addProperty("orderId", orderId);
        return client.post("/hostedOrderInfoFirst", data, JsonObject.class);
    }

    public JsonObject createHostedOrderDeposit(JsonObject data) throws APIError, IOException {
        return client.post("/createHostedOrderDeposit", data, JsonObject.class);
    }

    public JsonObject getHostedCoinUSDTPrice(com.google.gson.JsonArray coinIds) throws APIError, IOException {
        JsonObject data = new JsonObject();
        data.add("coinIds", coinIds);
        return client.post("/getHostedCoinUSDTPrice", data, JsonObject.class);
    }

    public JsonObject getMainCoinList() throws APIError, IOException {
        return client.post("/getMainCoinList", null, JsonObject.class);
    }

    public JsonObject createAppDemoOrderDeposit(JsonObject data) throws APIError, IOException {
        return client.post("/createAppDemoOrderDeposit", data, JsonObject.class);
    }

    public void confirmTrade(JsonObject data) throws APIError, IOException {
        client.post("/confirmTrade", data, Void.class);
    }
}
